themes/default/content/blog/amazon-s3-express-one-zone/index.md

```diff
@@ -0,0 +1,279 @@
+
+```bash
+$ pulumi up
+     Type                          Name                        Status
+ +   pulumi:pulumi:Stack           s3-express-one-zone-py-dev  created (4s)
+ +   ├─ random:index:RandomString  bucket-name                 created (0.09s)
+ +   ├─ aws:s3:DirectoryBucket     my-bucket                   created (2s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file2.txt           created (1s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file3.txt           created (1s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file1.txt           created (1s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file4.txt           created (1s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file5.txt           created (1s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file7.txt           created (2s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file8.txt           created (2s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file9.txt           created (2s)
+ +   ├─ aws:s3:BucketObjectv2      my-file-file0.txt           created (3s)
+ +   └─ aws:s3:BucketObjectv2      my-file-file6.txt           created (3s)
+
+Outputs:
+    bucketID: "my-bucket-z5tn74--usw2-az1--x-s3"
+
+Resources:
+    + 13 created
+
+Duration: 11s
+```
+
+And there we go -- our new Amazon S3 Express One Zone bucket is available and its ID has been printed. We can check out its contents:
+
+```bash
+$ aws s3api list-objects-v2 \
+    --region=us-west-2 \
+    --bucket $(pulumi stack output bucketID)
+```
+
+Note that the storage class of these objects is listed as `EXPRESS_ONEZONE`:
+
+```
+{
+    "Contents": [
+        {
+            "Key": "my-file-file6.txt",
+            "LastModified": "2023-12-03T17:18:06+00:00",
+            "ETag": "\"1254d05e685245ef9fdcca10f3c547bc\"",
+            "ChecksumAlgorithm": [
+                "CRC32"
+            ],
+            "Size": 102400,
+            "StorageClass": "EXPRESS_ONEZONE"
+        },
+        ...
+        {
+            "Key": "my-file-file2.txt",
+            "LastModified": "2023-12-03T17:18:04+00:00",
+            "ETag": "\"d5310d3efd394c96bf5bf4379fc81bff\"",
+            "ChecksumAlgorithm": [
+                "CRC32"
+            ],
+            "Size": 102400,
+            "StorageClass": "EXPRESS_ONEZONE"
+        }
+    ]
+}
+```
+
+We can also see our bucket and its contents in the S3 Management Console:
+
+![Amazon S3 Express One Zone console screenshot](s3-console-express.png)
+
+General-purpose buckets and directory buckets have different tabs. There are also utilities available to import data from other buckets and perform batch operations.
+
+## Benchmarking Express One Zone Performance
+
+To demonstrate how much faster Express One Zone buckets are to access, let's run a few benchmarks. To do this, we set up two buckets -- an Express One Zone and another using Standard storage class -- along with an EC2 instance in the same AZ as our One Zone bucket.
```

Comment:
Especially fun if the program includes the benchmark in userdata or remote Command - and writes resutls to S3 as well :-).  So that the entire thing is in Pulumi!

Reply:
Unfortunately, did not have the foresight to go this route. I agree it would have been nice. Given the time constraints around the post, and that this isn't essential to the core, going to have to skip this one for now.